docs/zed/README.md

```diff
@@ -102,198 +292,357 @@ scans that run in the opposing order.
+range scans over such data is impaired.
+
+### 1.5 Time Travel
+
+Because commits are transactional and immutable, a query
+sees its entire data scan as a fixed "snapshot" with respect to the
+commit history.  In fact, Zed's [from operator](../zq/operators/from.md)
+allows a commit object to be specified with the `@` suffix to a
+pool reference, e.g.,
+```
+zed query 'from logs@1tRxi7zjT7oKxCBwwZ0rbaiLRxb | ...'
+```
+In this way, a query can time-travel through the commit history.  As long as the
+underlying data has not been deleted, arbitrarily old snapshots of the Zed
+lake can be easily queried.
+
+If a writer commits data after and while a reader is scanning, then the reader
+does not see the new data since it's scanning the snapshot that existed
+before these new writes occurred.
+
+Also, arbitrary metadata can be committed to the log as described below,
+e.g., to associate index objects or derived analytics to a specific
+journal commit point potentially across different data pools in
+a transactionally consistent fashion.
+
+While time travel through commit history provides one means to explore
+past snapshots of the commit history, another means is to use a timestamp.
+Because the entire history of branch updates is stored in a transaction journal
+and each entry contains a timestamp, branch references can be easily
+navigated by time.  For example, a list of branches of a pool's past
+can be created by scanning the branches log and stopping at the largest
+timestamp less than or equal to the desired timestamp.  Likewise, a branch
+can be located in a similar fashion, then its corresponding commit object
+can be used to construct that data of that branch at that past point in time.
+
+ > Note that time travel using time stamps is a forthcoming feature.
+
+### 1.6 Search Indexes
+
+Unlike traditional indexing systems based on an inverted-keyword index,
+indexing in Zed is decentralized and incremental.  Instead of rolling up
+index data structures across many data objects, a Zed lake stores a small
+amount of index state for each data object.  Moreover, the design relies on
+indexes only to enhance performance, not to implement the data semantics.
+Thus, indexes need not exist to operate a lake and can be incrementally added or
+deleted without large indexing jobs needing to rebuild a monolithic index
+after each configuration change.
+
+To optimize pool scans, the lake design relies on the well-known pruning
+concept to skip any data object that the planner determines can be skipped
+based on one or more indexes of that object.  For example, if an object
+has been indexed for field "foo" and the query
+```
+foo == "bar" | ...
+```
+is run, then the scan will consult the "foo" index and skip the data object
+if the value "bar" is not in that index.
+
+Also, each data object is broken up into seekable chunks and the chunk location
+of each index value is stored in the index so often only parts of large
```

Comment:
@mccanne: I only suggested that change because I find the words "so often" can disrupt the flow of a sentence, since I see it frequently used to say things like "it happens so often!" No offense taken if that change is rejected.